gradle/compatibility.gradle

```diff
@@ -0,0 +1,176 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+buildscript {
+  repositories {
+    mavenCentral()
+    jcenter()
+  }
+  dependencies {
+    classpath "org.ajoberstar:grgit:1.9.3"
+    classpath "com.github.siom79.japicmp:japicmp:0.13.0"
+  }
+}
+
+import japicmp.cli.JApiCli
+import org.ajoberstar.grgit.Grgit
+import org.apache.tools.ant.DirectoryScanner
+
+// Parameters
+def baseBranch = project.hasProperty('baseBranch') ? project.baseBranch : "trunk"
+def newBranch = project.hasProperty('newBranch') ? project.newBranch : "HEAD"
+def onlyIncompatible = project.hasProperty('onlyIncompatible') ? project.onlyIncompatible : true
+def failOnBinaryIncompatibility = project.hasProperty('failOnBinaryIncompatibility') ?
+        project.failOnBinaryIncompatibility : false
+def failOnSourceIncompatibility = project.hasProperty('failOnSourceIncompatibility') ?
+        project.failOnSourceIncompatibility : false
+def failOnModifications = project.hasProperty('failOnModifications') ?
+        project.failOnModifications : false
+def failOnSemanticIncompatibility = project.hasProperty('failOnSemanticIncompatibility') ?
+        project.failOnSemanticIncompatibility : false
+
+// Constants
+def packageIncludes = [
+  "org.apache.kafka.common.*",
+  "org.apache.kafka.clients.*",
+  "org.apache.kafka.connect.*",
+]
+def packageExcludes = [
+  "org.apache.kafka.common.internals.*",
+  "org.apache.kafka.clients.consumer.internals.*",
+  "org.apache.kafka.clients.producer.internals.*",
+  "org.apache.kafka.clients.common.internals.*",
+]
+def baseWorkspace = "$buildDir/tmp/compatibility/base"
+def newWorkspace = "$buildDir/tmp/compatibility/new"
+def reportDir = "$buildDir/reports/compatibility"
+def reportFile = "$reportDir/compatibilityReport.html"
+def rootRepo = Grgit.open(dir: rootProject.rootDir)
+
+def repoExcludes = new ArrayList<String>(rootRepo.clean(ignore: false, directories: true, dryRun: true))
+
+static def collectArtifactJars(String workspace) {
+  String includesPattern = '**/build/libs/kafka-*.jar'
```

Comment:
I think this does not take up connect artifacts. Those are called `connect-*.jar`. Is this intentional? If not, you might only want to add `...api` package, if I understand package structure there correctly.

Reply:
Good point, I've refined this list too. Excluded the examples and system-tests jars and included the connect ones. I think now it is good.